shell/plugins/codemirror.js

```diff
@@ -35,8 +31,22 @@ import 'codemirror/addon/hint/anyword-hint.js';
 
 import { strPad } from '@shell/utils/string';
 
-Vue.use(VueCodemirror);
-export default VueCodemirror;
+const vCodemirror = {
+  install: (Vue) => {
+    if (Vue.component('codemirror')) {
+      // eslint-disable-next-line no-console
+      console.debug('Skipping codemirror install. Component already exists.');
+    } else {
+      Vue.component('codemirror', vueCodemirror);
+    }
+  }
+};
+
+export default vCodemirror;
+/* eslint-disable-next-line no-console */
+console.warn('The implicit addition of vue-codemirror has been deprecated in Rancher Shell and will be removed in a future version. Make sure to invoke `Vue.use(vCodemirror)` to maintain compatibility.');
+
+Vue.use(vCodemirror);
```

Comment:
We will want to update these lines, similar to https://github.com/rancher/dashboard/pull/10943#discussion_r1589651500

Reply:
didn't you say before that these might also be used by extensions?

Reply:
That's correct, and I might have misattributed the issue to loading extensions. 

Based on the changes made to the i18n plugin[^1], I understand that the issue only has to do with loading our vue plugins for tests. @codyrancher is that an accurate assessment?

[^1]: https://github.com/rancher/dashboard/pull/10923/files#diff-4964064b3ea3d66459e9eb386e2612aa8aea93c2d858eca54128348b498dcabb